src/contexts/AuthContext.js

```diff
@@ -0,0 +1,56 @@
+import { createContext, useContext, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { loginService, signupService } from "toolkit/utils";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [auth, setAuth] = useState(() => {
+    const token = localStorage.getItem("token");
+    if (token) return { token, isAuth: true };
+    return { token: "", isAuth: false };
+  });
+  const [user, setUser] = useState(() => JSON.parse(localStorage.getItem("user")) ?? "");
+
+  const navigate = useNavigate();
+
+  const loginHandler = async (e, email, password) => {
+    e.preventDefault();
+    const [token, userData] = await loginService("/api/auth/login", email, password);
+    setUser(userData);
+    localStorage.setItem("token", token);
+    localStorage.setItem("isAuth", true);
+    localStorage.setItem("user", JSON.stringify(userData));
```

Comment:
Hey, why are you saving the userData in the localStorage? How are you going to use it?

Reply:
On refreshing, the user data was getting deleted. I thought of displaying user data like name, email in profile page if user has logged in by saving it in localStorage.

Reply:
What happens when a new user logs in? The old userData will get overwritten right? Also, animesh has implemented a better approach for "remember me" approach, where upon refresh the data won't be deleted.

Reply:
Yes, old data will be deleted from localStorage by logout function.
Oh nice, I'll definitely check that out!